ios/Codeforces Watcher/Application/features/news/NewsViewController.swift

```diff
@@ -1,102 +1,128 @@
-//
-//  NewsViewController.swift
-//  Codeforces Watcher
-//
-//  Created by Den Matyash on 12/31/19.
-//  Copyright © 2019 xorum.io. All rights reserved.
-//
-
-import UIKit
-import TinyConstraints
-import WebKit
+import SwiftUI
 import common
-import FirebaseAnalytics
-
-class NewsViewController: UIViewControllerWithFab, ReKampStoreSubscriber {
 
-    private let tableView = UITableView()
-    private let tableAdapter = NewsTableViewAdapter()
-    private let refreshControl = UIRefreshControl()
+class NewsViewController: UIHostingController<NewsView>, ReKampStoreSubscriber {
     
-    override func viewDidLoad() {
-        super.viewDidLoad()
-
-        setupView()
-        setupTableView()
+    private lazy var fabButton = FabButtonViewController(name: "newsShareIcon")
+    
+    init() {
+        super.init(rootView: NewsView())
+        
+        setRefreshControl()
+        setInteractions()
     }
-
+    
+    @objc required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-
+        
+        hideNavigationBar()
+        setFabButton()
+        fabButton.show()
+        
         store.subscribe(subscriber: self) { subscription in
             subscription.skipRepeats { oldState, newState in
-                return KotlinBoolean(bool: oldState.news == newState.news)
+                KotlinBoolean(bool: oldState.news == newState.news)
             }.select { state in
-                return state.news
+                state.news
             }
         }
     }
-
+    
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
+        
+        fabButton.hide()
+        
         store.unsubscribe(subscriber: self)
     }
-
-    private func setupView() {
-        navigationController?.interactivePopGestureRecognizer?.isEnabled = false
-        view.backgroundColor = .white
-
-        buildViewTree()
-        setConstraints()
-        setFabImage(named: "shareImage")
+    
+    private func setFabButton() {
+        tabBarController?.tabBar.addSubview(fabButton.view)
+        fabButton.setView()
+        fabButton.setButtonAction(action: { self.onFabButton() })
     }
-
-    private func buildViewTree() {
-        view.addSubview(tableView)
+    
+    private func onFabButton() {
+        let activityController = UIActivityViewController(activityItems: ["share_cw_message".localized],
```

Comment:
The usual code style for such cases is to break line before the first argument, like this:

```
let activityController = UIActivityViewController(
    activityItems: ["share_cw_message".localized],
    applicationActivities: nil
).apply {
    // ...
}
```